src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -776,32 +786,36 @@ class ActionDelegate {
         return userService.loggedUser
     }
 
-    void generatePDF(String transitionId, String fileFieldId) {
+    void generatePDF(Transition t, FileField field) {
+        this.generatePDF(t.getStringId(), field.importId)
+    }
+
+    void generatePDF(String transitionId, String fileFieldId, Case fromCase = useCase, Case saveToCase = useCase) {
         PdfResource pdfResource = ApplicationContextProvider.getBean(PdfResource.class) as PdfResource
         String filename = pdfResource.getOutputDefaultName()
-        String storagePath = pdfResource.getOutputFolder() + File.separator + useCase.stringId + "-" + fileFieldId + "-" + pdfResource.getOutputDefaultName()
+        String storagePath = pdfResource.getOutputFolder() + File.separator + saveToCase.stringId + "-" + fileFieldId + "-" + pdfResource.getOutputDefaultName()
```

Comment:
the PDF methods seem very similar, which is amplified by the fact that the lines are displayed twice in the diff. In my opinion, at least the `storagePath` creation should be abstracted into a single method call in all implementations.

Reply:
Corrected